c7n/resources/apigw.py

```diff
@@ -742,10 +749,18 @@ class SetWafv2(BaseAction):
     permissions = ('wafv2:AssociateWebACL', 'wafv2:ListWebACLs')
 
     schema = type_schema(
-        'set-wafv2', required=['web-acl'], **{
```

Comment:
the logic for web-acl required probably should go into validate, based on if state: true.

Reply:
Added **PolicyValidation** check for state parameter.